util.js

```diff
@@ -251,6 +251,56 @@ var connectSmart = function() {
         return connectSmart3A.apply(null, arguments)
 }
 
+/**
+ * Helper class to wait for the system startup to be completed.
+ * Adding widgets before the desktop is ready to accept them can result in errors.
+ */
+var StartUpCompletionHelper = class AppIndicators_StartUpCompletionHelper {
+
+    constructor() {
+        this._startupComplete = false;
+        this._displayAvailable = false;
+        this._listeners = new Set();
+
+        // If the desktop is still starting up, we must wait until it is ready
+        if (Main.layoutManager._startingUp) {
+            let startupPreparedId = Main.layoutManager.connect('startup-complete', () => {
+                Main.layoutManager.disconnect(startupPreparedId);
+                this._startupComplete = true;
+                this._checkStartUpComplete();
+            });
+        } else {
+            this._startupComplete = true;
+        }
+
+        // Ensure that the default Gdk Screen is available
+        if (Gtk.IconTheme.get_default() == null) {
```

Comment:
i think we could use Strict Equality Comparison (===)

Reply:
Updated